drivers/hmis/app/graphql/mutations/delete_client.rb

```diff
@@ -20,6 +20,10 @@ def resolve(id:, client_lock_version: nil, confirmed: false)
       client = Hmis::Hud::Client.viewable_by(current_user).find_by(id: id)
       client.lock_version = client_lock_version if client_lock_version
 
+      # While this is redundant with the viewable_by() scope above, this check caches the authorization result so that
+      # the client object-level authorization check will succeed even after the client has been deleted
+      access_denied! unless current_permission?(permission: :can_view_clients, entity: client)
```

Comment:
Is `:can_view_clients` enough to allow deletion?

Reply:
No, `can_delete_clients` is required. But that is checked below on line 35.